relays/lib-substrate-relay/src/finality/target.rs

```diff
@@ -53,10 +53,20 @@ impl<P: SubstrateFinalitySyncPipeline> SubstrateFinalityTarget<P> {
 	pub async fn ensure_pallet_active(&self) -> Result<(), Error> {
 		let is_halted = self.client.storage_value(P::FinalityEngine::is_halted_key(), None).await?;
 		if is_halted.unwrap_or(false) {
-			Err(Error::BridgePalletIsHalted)
-		} else {
-			Ok(())
+			return Err(Error::BridgePalletIsHalted)
 		}
+
+		let is_initialized =
+			super::initialize::is_initialized::<P::FinalityEngine, P::SourceChain, P::TargetChain>(
+				&self.client,
+			)
+			.await
+			.map_err(|e| Error::Custom(e.to_string()))?;
```

Comment:
Why return `Error::Custom` here and just propagate the initial error on `is_halted` check ?

Reply:
Do you mean - why not return `BridgePalletIsHalted` here instead of `Error::Custom`? Because it may be any kind of issue, unrelated to the uninitialized pallet - e.g. transport error or decoding. We may add some additional info to the error though - e,g.`format!( "Failed to read pallet initialization state: {:?}", e)`

Reply:
> Do you mean - why not return BridgePalletIsHalted here instead of Error::Custom?

I mean why not `await?` which would propagate the error returned by `P::FinalityEngine::is_initialized()` as we do for `P::FinalityEngine::is_halted(&self.client).await?`.

> Because it may be any kind of issue, unrelated to the uninitialized pallet - e.g. transport error or decoding. We may add some additional info to the error though - e,g.format!( "Failed to read pallet initialization state: {:?}", e)

True, but shouldn't this information be included in the error type ? For example if we're unable to read the initialization state we would get a `RpcError` I think. I'm not sure how detailed it would be though. Anyway adding some additional info might be safer, but I think we should do it both for `is_halted` and for `is_initialized` checks.

Reply:
Ah, ok - got it now :) Yeah - seems like an unneeded conversion here.